File: src/main/java/com/tjh/riskfactor/controller/SectionController.java
package com.tjh.riskfactor.controller;

import lombok.RequiredArgsConstructor;

import org.springframework.web.bind.annotation.*;

import com.tjh.riskfactor.entity.form.Section;
import com.tjh.riskfactor.service.SectionService;
import static com.tjh.riskfactor.error.ResponseErrors.notFound;

@RestController
@RequiredArgsConstructor
public class SectionController {

    private final SectionService service;

    @GetMapping("/sections/{sid}")
    Section section(@PathVariable Integer sid) {
        return service.section(sid).orElseThrow(() -> notFound("section", sid.toString()));
    }

}

File: backend/src/main/kotlin/com/tjh/riskfactor/controller/SectionController.kt
package com.tjh.riskfactor.controller

import org.springframework.web.bind.annotation.*

import com.tjh.riskfactor.service.SectionService

@CrossOrigin
@RestController
class SectionController(private val service: SectionService) {

    @GetMapping("/sections/{sid}")
    fun section(@PathVariable sid: Int) = service.checkedFind(sid)

}
